src/Tizen.NUI.Components/Controls/Picker.cs

```diff
@@ -413,10 +429,76 @@ private void Initialize()
 
             Focusable = true;
             KeyEvent += OnKeyEvent;
+
+            Accessibility.Accessibility.Enabled += (s, e) => {
+                isAtspiEnabled = true;
+                AccessibilityEnabled();
+            };
+            Accessibility.Accessibility.Disabled += (s, e) => {
+                isAtspiEnabled = false;
+                HideItemsForAccessibility();
+            };
+            Accessibility.Accessibility.ScreenReaderEnabled += (s, e) => {
+                isScreenReaderEnabled = true;
+            };
+            Accessibility.Accessibility.ScreenReaderDisabled += (s, e) => {
+                isScreenReaderEnabled = false;
+            };
+
+            isAtspiEnabled = Accessibility.Accessibility.IsEnabled;
+            isScreenReaderEnabled = Accessibility.Accessibility.IsScreenReaderEnabled;
+
+        }
+
+        private void HideItemsForAccessibility()
+        {
+            if (loopEnabled)
+                for (int i = accessibilityHiddenStartIdx; i < (accessibilityHiddenStartIdx + scrollVisibleItems); i++)
+                    itemList[i].AccessibilityHidden = true;
+            else
+            {
+                if (currentValue > (middleItemIdx * 2))
+                    for (int i = accessibilityHiddenStartIdx; i < (accessibilityHiddenStartIdx + (maxValue - minValue)); i++)
+                        itemList[i].AccessibilityHidden = true;
+                else
+                    for (int i = accessibilityHiddenStartIdx; i < (accessibilityHiddenStartIdx + (maxValue - minValue + 1)); i++)
+                        itemList[i].AccessibilityHidden = true;
+            }
+        }
+
+        private void ShowItemsForAccessibility(int startIdx)
```

Comment:
What about introducing public property that would allow to control how many items on both sides of the middle item should be shown for Accessibility? Value 0 would mean that only a middle item is shown to Accessibility, 1 - will extend the range by 1 item on each side of middle item, etc. The default value of this property could be adjusted to show to Accessibility all visible items (what would match the implementation from this patch), but when needed (for example in date picker) the range of items available for accessibility could be narrowed.